cloudinary-core/src/main/java/com/cloudinary/Cloudinary.java

```diff
@@ -284,6 +284,31 @@ public String downloadMulti(String[] urls, Map options) throws IOException {
         return buildUrl(cloudinaryApiUrl("multi", options), params);
     }
 
+    /**
+     * Generates URL for executing "Download Folder" operation on Cloudinary site.
+     * 
+     * @param folderPath path of folder to generate download URL for
+     * @param options    optional, holds hints for URL generation procedure, see documentation for full list
+     * @return generated URL for downloading specified folder as ZIP archive
+     */
+    public String downloadFolder(String folderPath, Map options) throws UnsupportedEncodingException {
+        if (StringUtils.isEmpty(folderPath)) {
+            throw new IllegalArgumentException("Folder path parameter value is required");
+        }
+
+        Map adjustedOptions = new HashMap();
+        if (options != null) {
+            adjustedOptions.putAll(options);
+        }
+
+        adjustedOptions.put("prefixes", folderPath);
+
+        final Object resourceType = adjustedOptions.get("resource_type");
+        adjustedOptions.put("resource_type", resourceType != null ? resourceType : "all");
+
+        return downloadArchive(adjustedOptions, "zip");
```

Comment:
We shouldn't force zip format - we should take the format from the options if the user provided a format. Otherwise we can simply pass `null` to the download archive method (there's a server-side default, no need to send a format at all if the user didn't pass any).